ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -8232,6 +8232,14 @@ private void handleLineage(LoadTableDesc ltd, Operator output)
         Warehouse wh = new Warehouse(conf);
         tlocation = wh.getDefaultTablePath(db.getDatabase(tableDesc.getDatabaseName()),
             tName, tableDesc.isExternal());
+
+        if (destinationTable != null) {
+          Table tbl = db.getTranslateTableDryrun(destinationTable.getTTable());
```

Comment:
isn't it a bit too late to run this? why don't we run this right away when we are creating the `destinationTable` object?
so that we don't make the same mistake again-and-again?